nexus/db-queries/src/db/datastore/rack.rs

```diff
@@ -89,15 +91,12 @@ enum RackInitError {
 }
 type TxnError = TransactionError<RackInitError>;
```

Comment:
Seems like only `rack_create_recovery_silo` remains using this and could be given the same treatment as `rack_populate_service_records`?

Reply:
I've gone ahead and just plumbed through `RackInitError` here, though I've added a variant to identify transaction retry errors.